### PR TITLE
Fix/top dih

### PR DESCRIPTION
**Pull request by fgrunewald**
This fixes an important issue in the top-parser. Apparently GROMACS is able to expand a single dihedral directive given in the molecule into multiple ones based on dihedral types. In this PR we now read lists of type parameters instead of a single value and in the 'grompp' step they get replaced properly i.e. the new dihedrals are generated as well.

### CODE REVIEW

File: polyply/src/topology.py [hunk 1]
```diff
@@ -332,9 +355,22 @@ def gen_bonded_interactions(self):
...
+                            if not meta:
+                                meta = {}
+                            if idx > 0:
+                                new_interaction = Interaction(atoms=tuple(interaction.atoms),
+                                                              parameters=new_param,
+                                                              meta=meta)
+                                additional_interactions[inter_type].append(new_interaction)
+                            else:
+                                interaction.parameters[:] = new_param[:]
+                                interaction.meta.update(meta)
```
Comment (pckroon) [inline polyply/src/topology.py:368]: I've read this code wrong a few times, wrote a wrong suggestion, and am still a bit confused. 
Could you swap the if and else statements? And add some comments?

File: polyply/tests/test_topology.py [hunk 2]
```diff
@@ -627,7 +670,8 @@ def test_replace_types(lines, outcome):
         polyply.src.top_parser.read_topology(new_lines, top)
         top.gen_bonded_interactions()
         for inter_type in outcome:
-            assert top.molecules[0].molecule.interactions[inter_type] == outcome[inter_type]
+            for inter in top.molecules[0].molecule.interactions[inter_type]:
+                assert top.molecules[0].molecule.interactions[inter_type] == outcome[inter_type]
```
Comment (pckroon) [inline polyply/tests/test_topology.py:674]: This looks wrong, you execute the same assert multiple times

**Review by pckroon — CHANGES_REQUESTED** [review]
Oops, missed this one. 
Two small comments, looks good to me otherwise

File: polyply/tests/test_topology.py (hunk 2 above)
Comment (fgrunewald) [reply polyply/tests/test_topology.py:674]: Yep that seems strange indeed

**Review by pckroon — APPROVED** [review]
lgtm :)

**Pull request merged**